src/shared/components/FormComponents/CheckboxController/CheckboxController.types.ts

```diff
@@ -8,6 +8,7 @@ type FormCheckboxProps = {
   onCustomChange?: (event: ChangeEvent<HTMLInputElement>) => void;
   'data-testid'?: string;
   sxLabelProps?: SxProps;
+  defaultControllerValue?: boolean;
```

Comment:
There seems to be a lot of interference from both `CheckboxProps` and `UseControllerProps<T>` regarding the type of the `defaultValue` attribute. I couldn't get a "optional boolean" `defaultValue` attribute to register proper, even if I put `& { defaultValue?: boolean }` at the end of `export type InputControllerProps = ...`.

So I decided to add an `defaultControllerValue` attribute instead.

Also, `CheckboxController` actually only need very few of these props, and it spread the rest of props (including `defaultValue`) into an inner `<Checkbox />` component. But this `defaultControllerValue` need to be set on the `<Controller />` component instead. So I feel adding the explicit `defaultControllerValue` is the most clear and clean implementation, given what we have to work with.

Reply:
I think this may be one of those "it works either way" kind of thing. Because `defaultValue` is a prop on `Controller` and it evidently works. That said, I was in fact looking for a way to do the way you linked but I couldn't find any examples 😖. So I'll give this another stab and try to make this part consistently with how things are currently done.

Reply:
Also, I see the difference in approaches now. What I did was to tell the form/UI controller: "if there is no value for auto-assign, then use `true` as the default"; whereas the way we're currently doing things with what you linked is to create a new object with explicitly set default values.